cwms-data-api/src/main/java/cwms/cda/data/dao/LocationsDaoImpl.java

```diff
@@ -125,6 +135,12 @@ public List<Location> getLocations(String nameRegex, String unitSystem, String d
             whereCondition = whereCondition.and(AV_LOC.VERTICAL_DATUM.equalIgnoreCase(datum));
         }
 
+        if (ignoredKinds != null) {
+            for (String kind : ignoredKinds) {
```

Comment:
I would just use an "in clause" here instead. performance wise likely to be the same but the code would be cleaner.

Either way extra queries will be created based on the number of ignored elements, but more of the "sets" would map to each other at the bind level and JOOQ should be able to reasonably take care of it.